lib/web_ui/test/engine/view_embedder/embedding_strategy/custom_element_embedding_strategy_test.dart

```diff
@@ -121,4 +121,54 @@ void doTests() {
           reason: 'Should be injected `nextTo` the passed element.');
     });
   });
+
+  group('context menu', () {
+    setUp(() {
+      target = createDomElement('this-is-the-target');
+      domDocument.body!.append(target);
+      strategy = CustomElementEmbeddingStrategy(target);
+      strategy.initialize();
+    });
+
+    tearDown(() {
+      target.remove();
+    });
+
+    test('disableContextMenu and enableContextMenu can toggle the context menu', () {
+      // When the app starts, contextmenu events are not prevented.
+      DomEvent event = createDomEvent('Event', 'contextmenu');
+      expect(event.defaultPrevented, isFalse);
+      target.dispatchEvent(event);
+      expect(event.defaultPrevented, isFalse);
+
+      // Disabling the context menu causes contextmenu events to be prevented.
+      strategy.disableContextMenu();
+      event = createDomEvent('Event', 'contextmenu');
+      expect(event.defaultPrevented, isFalse);
+      target.dispatchEvent(event);
+      expect(event.defaultPrevented, isTrue);
```

Comment:
Another important test is to dispatch an event outside of `target` (either create a sibling element or just dispatch on `document.body`) and make sure that the event wasn't prevented.

Reply:
Great idea, thanks!